.github/workflows/windows.yml

```diff
@@ -17,87 +17,128 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
-  mingw:
-    runs-on: windows-2022
-    strategy:
-      matrix:
-        architecture: [x64, x86]
+  # 1) Simple MSVC build on latest image (Release)
+  msvc-latest:
+    runs-on: windows-latest
+    steps:
+      - uses: egor-tensin/setup-mingw@84c781b557efd538dec66bde06988d81cd3138cf
+        with:
+           version: 13.2.0
+           static: 0   # v4.2.2 (keep the hash!)
+       # --- missing checkout step ---
+      - name: Checkout repository                       # <-- this is what Niels meant
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683  # v4.2.2 pinned
 
+    # now the sources are present, so CMake can see them
+      - name: Configure CMake
+        run: cmake -S . -B build -G "MinGW Makefiles" -DJSON_BuildTests=On     
+      - name: Configure CMake
+        run: >
+          cmake -S . -B build -G "Visual Studio 17 2022"
+          -DJSON_BuildTests=ON
+      - name: Build with MSVC
+        run: cmake --build build --config Release --parallel 10
+
+  # 2) Simple MinGW build on latest image
+  mingw-latest:
+    runs-on: windows-latest
     steps:
-      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+      - uses: actions/checkout@v3
       - name: Set up MinGW
         uses: egor-tensin/setup-mingw@84c781b557efd538dec66bde06988d81cd3138cf # v2.2.0
-        with:
-          platform: ${{ matrix.architecture }}
-          version: 12.2.0 # https://github.com/egor-tensin/setup-mingw/issues/14
-      - name: Run CMake
-        run: cmake -S . -B build -G "MinGW Makefiles" -DCMAKE_BUILD_TYPE=Debug -DJSON_BuildTests=On
+      - name: Configure
+        run: cmake -S . -B build -G "MinGW Makefiles" -DJSON_BuildTests=On
       - name: Build
         run: cmake --build build --parallel 10
       - name: Test
-        run: cd build ; ctest -j 10 -C Debug --output-on-failure
+        run: |
+          cd build
+          ctest -j 10 --output-on-failure
 
-  msvc:
+  # 3) Full MSVC matrix
+  msvc-matrix:
+    runs-on: ${{ matrix.os }}
     strategy:
       matrix:
+        os: [windows-2019, windows-2022]
```

Comment:
Windows Server 2019 has been retired. The Windows Server 2019 image has been removed as of 2025-06-30.